src/lib.rs

```diff
@@ -118,6 +146,35 @@ impl Drop for Model {
 	}
 }
 
+impl Drop for Metadata {
+	fn drop(&mut self) {
+		unsafe {
+			ds::DS_FreeMetadata(self.metadata);
+		}
+	}
+}
+
+use std::fmt;
+impl fmt::Display for Metadata {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		unsafe {
+			let num_items = (*self.metadata).num_items;
+			let vec_metadata = &[(*self.metadata).items];
+			let r = {
+				let mut v = Vec::new();
+				for item in 0..num_items {
+					let ptr = (*vec_metadata[item as usize]).character;
```

Comment:
This is not working but I'm not able to find proper docs on how this should be dealt with, accessing the `items` field of `struct Metadata`.